ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/fhir/FHIRMortalityDependencyMaskingProvider.java

```diff
@@ -0,0 +1,244 @@
+    if (this.minYears >= 0) {
+      LocalDate birthDate = getDateFromNode(parent, BIRTHDATE_FIELD);
+      if (birthDate == null) {
+        // not sure the age of the person, therefore assume deceased indicators must be removed
+        remove = true;
+      } else {
+        // get the deceased date value as a date
+        LocalDate mortalityDate = getDateFromNode(parent, DATE_TIME_FIELD);
+        if (mortalityDate == null) {
+          // not sure when the person became deceased, compare the birthdate to the current date
+          LocalDate currentDate = LocalDate.now();
+          remove = isTooYoung(birthDate, currentDate);
+        } else {
+          remove = isTooYoung(birthDate, mortalityDate);
+        }
+      }
+    }
+    return remove;
+  }
+
+  protected boolean isTooYoung(LocalDate birthDate, LocalDate compareDate) {
+    long age = ChronoUnit.YEARS.between(birthDate, compareDate);
+    boolean young = age <= this.minYears;
+    return young;
+  }
+
+  protected LocalDate getDateFromNode(ObjectNode parent, String propertyName) {
+    LocalDate date = null;
+    JsonNode node = parent.get(propertyName);
+    if (node != null && !node.isNull() && node.isValueNode() && node.asText(null) != null) {
+      String stringValue = node.asText();
+      date = getDateFromString(stringValue, propertyName);
+    }
+    return date;
+  }
+
+  protected LocalDate getDateFromString(String inStringValue, String propertyName) {
+    LocalDate date = null;
+    // check for time component and strip it off if found (FHIR would allow it in deceasedDateTime)
+    int timeIndex = inStringValue.indexOf('T');
+    String stringValue = timeIndex > -1 ? inStringValue.substring(0, timeIndex) : inStringValue;
+    try {
+      String[] components = stringValue.split("-");
+      if (components.length < 3) {
+        // LocalDate.parse() doesn't handle dates that are just year or year/month
+        // LocalDate.parse() requires four digit year, so enforce here for consistency
+        if (components[0].length() != 4) {
+          throw new RuntimeException("invalid year");
+        }
+        int year = Integer.parseInt(components[0]);
+        if (year < 0 || year > 9999) {
+          throw new RuntimeException("invalid year");
+        }
+        int month;
+        if (components.length > 1) {
+          // LocalDate.parse() requires two digit month, so enforce here for consistency
+          if (components[1].length() != 2) {
+            throw new RuntimeException("invalid month");
+          }
+          month = Integer.parseInt(components[1]);
+          if (month < 1 || month > 12) {
+            throw new RuntimeException("invalid month)");
+          }
+        } else {
+          month = 1;
+        }
+        date = LocalDate.of(year, month, 1);
+      } else {
+        // there is enough data provided for LocaleDate to parse the value
+        date = LocalDate.parse(stringValue);
+      }
+    } catch (Exception e) {
```

Comment:
Is it possible to catch a more specific exception here?  Do we need to catch all exceptions?

Reply:
There are a few different types of unrelated exceptions possible here.  I could have listed them all in the catch, but I believe we would want to handle any exception even unforeseen ones in this same way, so Exception seemed appropriate.

Reply:
There are a few different Exception types possible from this block.  I could have listed those in the catch, but I believe we would want to catch any exceptions in this block, even unforeseen ones, in the same way so I thought Exception was appropriate here.